flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/metrics/ScalingMetrics.java

```diff
@@ -156,6 +156,8 @@ public static void computeLagMetrics(
         var pendingRecords = flinkMetrics.get(FlinkMetric.PENDING_RECORDS);
         if (pendingRecords != null) {
             scalingMetrics.put(ScalingMetric.LAG, pendingRecords.getSum());
+        } else {
+            scalingMetrics.put(ScalingMetric.LAG, 0.);
```

Comment:
Wouldn't this logic affect sources that do not report the pending records metric at all ?
Those currently would go through a different codepath in: https://github.com/apache/flink-kubernetes-operator/blob/main/flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/metrics/ScalingMetrics.java#L111

Reply:
I think both can share the same logic now. The linked code may be dead code now.

Reply:
Ok, in that case can you please remove the parts that are not necessary anymore?

Reply:
I'll follow up with a cleanup PR.